.github/workflows/dev-cmake.yml

```diff
@@ -0,0 +1,89 @@
+    inputs:
+      file_base:
+        description: "The common base name of the source tarballs"
+        required: true
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  build_and_test:
+    name: "CMake Developer Mode Express Test Workflows"
+
+    # Don't run the action if the commit message says to skip CI
+    if: "!contains(github.event.head_commit.message, 'skip-ci')"
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set file base name (Windows)
+        id: set-file-base
+        run: |
+           FILE_NAME_BASE=$(echo "${{ inputs.file_base }}")
+           echo "FILE_BASE=$FILE_NAME_BASE" >> $GITHUB_OUTPUT
+
+      - name: Install Linux Dependencies
+        run: |
+           sudo apt-get update
+           sudo apt-get install ninja-build doxygen graphviz
+           sudo apt install libssl3 libssl-dev libcurl4 libcurl4-openssl-dev
+           sudo apt install libaec0 libaec-dev
+           sudo apt install gcc-12 g++-12 gfortran-12
+           echo "CC=gcc-12" >> $GITHUB_ENV
+           echo "CXX=g++-12" >> $GITHUB_ENV
+           echo "FC=gfortran-12" >> $GITHUB_ENV
+
+      # Get files created by release script
+      - name: Get tgz-tarball (Linux)
+        uses: actions/download-artifact@cc203385981b70ca67e1cc392babf9cc229d5806 # v4.1.9
+        with:
+              name: tgz-tarball
+              path: ${{ github.workspace }}
+
+      - name: List files for the space (Linux)
+        run: |
+              ls -l ${{ github.workspace }}
+              ls ${{ runner.workspace }}
+
+      - name: Uncompress source (Linux)
+        run: tar -zxvf ${{ github.workspace }}/${{ steps.set-file-base.outputs.FILE_BASE }}.tar.gz
+
+      - name: List files for the space (Linux)
+        run: |
+              ls -l ${{ github.workspace }}
+              ls ${{ runner.workspace }}
+
+      - name: CMake Configure
+        shell: bash
+        run: |
+            mkdir "${{ runner.workspace }}/build"
+            cd "${{ runner.workspace }}/build"
+            cd ""
+            cmake -C ${{ github.workspace }}/hdfsrc/config/cmake/cacheinit.cmake \
+                 -G Ninja \
+                 -DCMAKE_BUILD_TYPE=Developer \
+                 -DBUILD_SHARED_LIBS=ON \
+                 -DHDF5_ENABLE_ALL_WARNINGS=ON \
+                 -DHDF5_ENABLE_PARALLEL:BOOL=OFF \
+                 -DHDF5_BUILD_CPP_LIB:BOOL=OFF \
+                 -DHDF5_BUILD_FORTRAN=OFF \
+                 -DHDF5_BUILD_JAVA=OFF \
+                 -DHDF5_BUILD_DOC=OFF \
+                 -DLIBAEC_USE_LOCALCONTENT=OFF \
+                 -DZLIB_USE_LOCALCONTENT=OFF \
+                 -DHDF_TEST_EXPRESS=0 \
+                 ${{ github.workspace }}/hdfsrc
+
+      - name: CMake Build
+        run: cmake --build . --parallel 3 --config Developer
+        working-directory: ${{ runner.workspace }}/build
+
+      - name: CMake Run Tests
+        env:
+          HDF5TestExpress: 0
+        run: ctest . --parallel 2 -C Developer -V -R H5TESTXPR
```

Comment:
Why not run all the tests? The testexpress tests aren't the only ones that would be affected.

Reply:
If there's a compile issue we should probably fix that. But if I remember right, the H5TESTXPR category doesn't cover many tests at all. Most of the libraries tests don't even pay attention to TestExpress.